src/Controls/src/Core/Platform/Android/Extensions/SemanticExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿#nullable enable
+
+using AndroidX.Core.View;
+using AndroidX.Core.View.Accessibility;
+using AView = Android.Views.View;
+
+namespace Microsoft.Maui.Controls.Platform
+{
+	public static partial class SemanticExtensions
+	{
+		public static void UpdateSemanticNodeInfo(this View virtualView, AccessibilityNodeInfoCompat? info)
+		{
+			if (info == null)
+				return;
+
+			if(virtualView.TapGestureRecognizerNeedsDelegate())
+				info.AddAction(AccessibilityNodeInfoCompat.AccessibilityActionCompat.ActionClick);
+		}
+
+		internal static void ApplyControlsAccessibilityDelegateIfNeeded(this View virtualView)
```

Comment:
consider renaming - at the very least, maybe we can drop the suffix and just call it `ApplyControlsAccessibilityDelegate`? I feel like your comment below suffices. But if not, then maybe `ApplyControlsAccessibilityDelegateIfPresent`?

Reply:
See following comment first (I'm assuming this naming is related to `TapGestureRecognizerNeedsDelegate` which doesn't fully make sense to me)